app/src/main/java/ch/epfl/favo/favor/Favor.java

```diff
@@ -70,9 +71,33 @@ public Favor createFromParcel(Parcel in) {
   public Favor() {}
 
   public Favor(
-      String title, String description, String requesterId, FavoLocation location, Status statusId) {
+      String title,
+      String description,
+      String requesterId,
+      FavoLocation location,
+      Status statusId) {
     String id = DatabaseWrapper.generateRandomId();
-    setId(id);
+    setParameters(id, title, description, requesterId, location, statusId);
+  }
+
+  public Favor( // includes id
+      String id,
+      String title,
+      String description,
+      String requesterId,
+      FavoLocation location,
+      Status statusId) {
+    setParameters(id, title, description, requesterId, location, statusId);
```

Comment:
I would avoid creating a method to set parameters, you can just have two constructors with a different signature and, since one is a subset of the other, use them directly.

Reply:
updated